src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -2302,11 +2336,14 @@ Status ColorControlServer::moveColorCommand(EndpointId endpoint, const Commands:
  * @param commandData Struct containing the parameters of the command
  * @return Status::Success when successful,
  *         Status::InvalidCommand when a step X and Y of 0 is provided
- *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a Color XY transition state.
+ *         Status::UnsupportedEndpoint when the provided endpoint doesn't correspond with a Color XY transition state,
+ *         Status::ConstraintError when a command parameters is outside its defined value range.
  */
 Status ColorControlServer::stepColorCommand(EndpointId endpoint, const Commands::StepColor::DecodableType & commandData)
 {
     VerifyOrReturnValue(commandData.stepX != 0 || commandData.stepY != 0, Status::InvalidCommand);
+    // Command Parameters constraint checks:
+    VerifyOrReturnValue(commandData.transitionTime <= kMaxtransitionTime, Status::ConstraintError);
```

Comment:
This check should be before the stepX/stepY != 0 checks, per spec, since constraint checks happen before prose.

Please check the other callsites in this PR for similar issues.